lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class Alerts {
+	/**
+	 * The alerts service address
+	 *
+	 * @var string
+	 */
+	protected $service_address;
+
+	/**
+	 * The alerts service port
+	 *
+	 * @var string
+	 */
+	protected $service_port;
+
+	/**
+	 * Full path to the alerts service
+	 *
+	 * @var string
+	 */
+	protected $service_url;
+
+	/**
+	 * Instance of his Alerts class
+	 *
+	 * @var Alerts
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Contructor
+	 *
+	 * Set to protected to prevent direct instantiation.
+	 * Use Alerts::get_instance() to get an instance of Alerts
+	 */
+	protected function __construct() {
+		if ( ! defined( 'ALERT_SERVICE_ADDRESS' ) || ! ALERT_SERVICE_ADDRESS ) {
+			throw new Exception( 'Missing alerts service host configuration in ALERT_SERVICE_ADDRESS constant' );
+		}
+
+		if ( ! defined( 'ALERT_SERVICE_PORT' ) || ! ALERT_SERVICE_PORT ) {
+			throw new Exception( 'Missing alerts service port configuration in ALERT_SERVICE_PORT constant' );
+		}
+
+		$this->service_address = ALERT_SERVICE_ADDRESS;
+		$this->service_port = ALERT_SERVICE_PORT;
+		$this->service_url = sprintf( 'http://%s:%s/v1.0/alerts', $this->service_address, $this->service_port );
+	}
+
+	/**
+	 * Send the alert
+	 *
+	 * @param array $body The alert message body
+	 *
+	 * @return array Response details from wp_remote_post
+	 */
+	protected function send( $body ) {
+		$response = wp_remote_post( $this->service_url, [
+			'timeout' => 0.1,
+			'body' => json_encode( $body ),
+		] );
+
+		if ( is_wp_error( $response ) ) {
+			throw new Exception( sprintf( 'Error sending alert: %s', $response->get_error_message() ) );
```

Comment:
We should definitely not throw an exception here - we need to return the WP_Error.

Reply:
My thinking is to have the main `Alerts::chat()` method to catch and handle all the exceptions just so that we have a more convenient and consistent way of logging them.

Reply:
OK but the point is, this function could - in theory - be called by functions other than `Alerts::chat()` in a future, and when that time comes, we would be breaking WP coding standards and introducing additional headaches.
I would prefer sticking to the normal way to do things even though it does not seem vital for the sake of the current PR: it could become an issue in the future.